### PR TITLE
Updated documentation for OshiSysMonitor

### PR DESCRIPTION
**Pull request by hardikbajaj**
Fixes #14359

### Description
Updates documentation for OshiSysMontior

#### Release note

<hr>

##### Key changed/added classes in this PR

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by hardikbajaj** [comment]
I didn't noticed wrong formatter formatted this file on my local :( . I'll get back with updated documentation

File: docs/operations/metrics.md [hunk 1]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
 |`ingest/segments/count`|Count of final segments created by job (includes tombstones). | `dataSource`, `taskId`, `taskType`, `groupId`, `taskIngestionMode`, `tags` |At least `1`.|
 |`ingest/tombstones/count`|Count of tombstones created by job. | `dataSource`, `taskId`, `taskType`, `groupId`, `taskIngestionMode`, `tags` |Zero or more for replace. Always zero for non-replace tasks (always zero for legacy replace, see below).|
 
-The `taskIngestionMode` dimension includes the following modes: 
-* `APPEND`: a native ingestion job appending to existing segments 
+The `taskIngestionMode` dimension includes the following modes:
+
```
[the remark this answers is not in the document]
Comment (hardikbajaj) [reply docs/operations/metrics.md:169]: This blank line is coming again with auto formatting. Do you want me to remove this ?

File: docs/operations/metrics.md [hunk 2]
```diff
@@ -23,7 +23,8 @@ title: "Metrics"
   -->
 
 
-You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
+You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring
```
Comment (ektravel) [inline docs/operations/metrics.md:26]:
```suggestion
You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
```

File: docs/operations/metrics.md [hunk 3]
```diff
@@ -23,7 +23,8 @@ title: "Metrics"
...
+query execution, ingestion, coordination, and so on.
```
Comment (ektravel) [inline docs/operations/metrics.md:27]: Remove line 27.

File: docs/operations/metrics.md [hunk 4]
```diff
@@ -40,7 +41,8 @@ Metrics may have additional dimensions beyond those listed above.
 ## Query metrics
 
 ### Router
-|Metric|Description|Dimensions|Normal value|
+
+|Metric|Description|Dimensions|Normal Value|
```
Comment (ektravel) [inline docs/operations/metrics.md:45]:
```suggestion
|Metric|Description|Dimensions|Normal value|
```

File: docs/operations/metrics.md [hunk 5]
```diff
@@ -135,7 +137,8 @@ Metrics may have additional dimensions beyond those listed above.
 
 #### Memcached only metrics
 
-Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to from the cache retrieval layer.
+Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to
```
Comment (ektravel) [inline docs/operations/metrics.md:140]:
```suggestion
Memcached client metrics are reported as per the following. These metrics come directly from the client.
```

File: docs/operations/metrics.md [hunk 6]
```diff
@@ -135,7 +137,8 @@ Metrics may have additional dimensions beyond those listed above.
...
+from the cache retrieval layer.
```
Comment (ektravel) [inline docs/operations/metrics.md:141]: Remove line 141.

File: docs/operations/metrics.md [hunk 7]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
 of the `isAppendToExisting` and `isDropExisting` flags in the
 task's `IOConfig` as follows:
 
-|`isAppendToExisting`|`isDropExisting`|Mode|
-|--------------------|----------------|----|
-|`true`|`false`|`APPEND`|
-|`true`|`true `|Invalid combination, exception thrown.|
-|`false`|`false`|`REPLACE_LEGACY`. The default for JSON-based batch ingestion. |
-|`false`|`true`|`REPLACE`|
+| `isAppendToExisting` | `isDropExisting` | mode |
```
Comment (ektravel) [inline docs/operations/metrics.md:178]:
```suggestion
|`isAppendToExisting`|`isDropExisting`|mode|
```

File: docs/operations/metrics.md [hunk 8]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
+|----------------------|-------------------|------|
```
Comment (ektravel) [inline docs/operations/metrics.md:179]:
```suggestion
|--------------------|----------------|----|
```

File: docs/operations/metrics.md [hunk 9]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
+| `true`               | `false` | `APPEND`|
```
Comment (ektravel) [inline docs/operations/metrics.md:180]:
```suggestion
|`true`|`false`|`APPEND`|
```

File: docs/operations/metrics.md [hunk 10]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
+| `true`               | `true  ` | Invalid combination, exception thrown. |
```
Comment (ektravel) [inline docs/operations/metrics.md:181]:
```suggestion
|`true`|`true`|Invalid combination, exception thrown.|
```

File: docs/operations/metrics.md [hunk 11]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
+| `false`              | `false` | `REPLACE_LEGACY` (this is the default for native batch ingestion). |
```
Comment (ektravel) [inline docs/operations/metrics.md:182]:
```suggestion
|`false`|`false`|`REPLACE_LEGACY` (This is the default for native batch ingestion.)|
```

File: docs/operations/metrics.md [hunk 12]
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
...
+| `false`              | `true` | `REPLACE`|
```
Comment (ektravel) [inline docs/operations/metrics.md:183]:
```suggestion
|`false`|`true`|`REPLACE`|
```

File: docs/operations/metrics.md [hunk 13]
```diff
@@ -239,9 +243,10 @@ batch ingestion emit the following metrics. These metrics are deltas for each em
 |`ingest/notices/queueSize`|Number of pending notices to be processed by the coordinator.|`dataSource`, `tags`|Typically 0 and occasionally in lower single digits. Should not be a very high number. |
 |`ingest/notices/time`|Milliseconds taken to process a notice by the supervisor.|`dataSource`, `tags`| < 1s |
 |`ingest/pause/time`|Milliseconds spent by a task in a paused state without ingesting.|`dataSource`, `taskId`, `tags`| < 10 seconds|
-|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
-|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
-If the JVM does not support CPU time measurement for the current thread, `ingest/merge/cpu` and `ingest/persists/cpu` will be 0.
+|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on coordinator cycle time.|
```
Comment (ektravel) [inline docs/operations/metrics.md:246]:
```suggestion
|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
```

File: docs/operations/metrics.md [hunk 14]
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
 
 ## Shuffle metrics (Native parallel task)
 
-The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`
+The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor`
```
Comment (ektravel) [inline docs/operations/metrics.md:287]:
```suggestion
The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`.
```

File: docs/operations/metrics.md [hunk 15]
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
...
+in `druid.monitoring.monitors`
```
Comment (ektravel) [inline docs/operations/metrics.md:288]: Remove line 288.

Comment (ektravel) [inline docs/operations/metrics.md:288]:
```suggestion
in `druid.monitoring.monitors`.
```

File: docs/operations/metrics.md [hunk 16]
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
...
 See [Enabling Metrics](../configuration/index.md#enabling-metrics) for more details.
```
Comment (ektravel) [inline docs/operations/metrics.md:289]:
```suggestion
See [Enabling metrics](../configuration/index.md#enabling-metrics) for more details.
```

File: docs/operations/metrics.md [hunk 17]
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
 |`ingest/events/buffered`|Number of events queued in the `EventReceiverFirehose` buffer.|`serviceName`, `dataSource`, `taskId`, `taskType`, `bufferCapacity`|Equal to the current number of events in the buffer queue.|
 |`ingest/bytes/received`|Number of bytes received by the `EventReceiverFirehose`.|`serviceName`, `dataSource`, `taskId`, `taskType`|Varies|
 
-## Sys
+## Sys [Deprecated]
+
+> SysMonitor will be deprecated due to its reliance on the sigar library, which is no longer maintained.
```
Comment (ektravel) [inline docs/operations/metrics.md:413]:
```suggestion
> SysMonitor will be deprecated due to its reliance on the Sigar library, which is no longer maintained.
```

File: docs/operations/metrics.md [hunk 18]
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
...
+> A new Monitor called [OshiSysMonitor](#oshisysmonitor), which provides similar functionality but is based on the oshi
```
Comment (ektravel) [inline docs/operations/metrics.md:414]:
```suggestion
> Instead, use the new OSHI monitor called [OshiSysMonitor](#oshisysmonitor). The new monitor has a wider support for different machine architectures including ARM instances.
```

File: docs/operations/metrics.md [hunk 19]
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
...
+> library. We encourage all users to switch to OshiSysMonitor as soon as possible.
```
Comment (ektravel) [inline docs/operations/metrics.md:415]: Remove line 415.

File: docs/operations/metrics.md [hunk 20]
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
...
-|`sys/disk/read/size`|Bytes read from disk. One indicator of the amount of paging occurring for segments.|`fsDevName`,`fsDirName`, `fsTypeName`, `fsSysTypeName`, `fsOptions`|Varies|
-|`sys/net/write/size`|Bytes written to the network|`netName`, `netAddress`, `netHwaddr`|Varies|
-|`sys/net/read/size`|Bytes read from the network|`netName`, `netAddress`, `netHwaddr`|Varies|
-|`sys/fs/used`|Filesystem bytes used|`fsDevName`, `fsDirName`, `fsTypeName`, `fsSysTypeName`, `fsOptions`|< max|
-|`sys/fs/max`|Filesystem bytes max|`fsDevName`, `fsDirName`, `fsTypeName`, `fsSysTypeName`, `fsOptions`|Varies|
-|`sys/mem/used`|Memory used||< max|
-|`sys/mem/max`|Memory max||Varies|
-|`sys/storage/used`|Disk space used|`fsDirName`|Varies|
-|`sys/cpu`|CPU used|`cpuName`, `cpuTime`|Varies|
+| Metric                 | Description                                                                          |Dimensions|Normal Value|
```
Comment (ektravel) [inline docs/operations/metrics.md:419]: If `SysMonitor` is deprecated, should these metrics be removed?

Also, please fix the table formatting. Do not add whitespace to make columns longer. Trim excessive white space between columns.

File: docs/operations/metrics.md [hunk 21]
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
...
+| `sys/mem/used`         | Memory used                                                                          ||< max|
+| `sys/mem/max`          | Memory max                                                                           ||Varies|
+| `sys/storage/used`     | Disk space used                                                                      |`fsDirName`|Varies|
+| `sys/cpu`              | CPU used                                                                             |`cpuName`, `cpuTime`|Varies|
+
+## OshiSysMonitor
+
+These metrics are only available if the `OshiSysMonitor` module is included.
+
+| Metric                     | Description                                                                          | Dimensions                           | Normal Value        |
```
Comment (ektravel) [inline docs/operations/metrics.md:442]: Please fix the table formatting.

Do not add whitespace to make columns longer. Trim excessive white space between columns.

File: website/.spelling [hunk 22]
```diff
@@ -2033,6 +2033,7 @@ org.apache.druid
 org.apache.druid.jetty.RequestLog
 org.apache.hadoop
 overlord.html
+oshi
```
Comment (ektravel) [inline website/.spelling:2036]:
```suggestion
OSHI
```

File: website/.spelling [hunk 23]
```diff
@@ -2051,6 +2052,7 @@ slf4j
 sql
 sqlQuery
 successfulSending
+sigar
```
Comment (ektravel) [inline website/.spelling:2055]:
```suggestion
[S]igar
```

**Review by ektravel — COMMENTED** [review]
Reviewed documentation portion of the PR.

**Comment by hardikbajaj** [comment]
@ektravel @abhishekagarwal87 , This PR got closed accidentally, I have moved it to a new PR now https://github.com/apache/druid/pull/14912 and addressed all the comments there.

**Pull request closed without merging**